app/models/user.rb

```diff
@@ -31,6 +31,22 @@ def invalidate_all_sessions!
 
   field :role, type: String
 
+  def in_agency_group?
+    %w[agency agency_with_refund agency_super].include?(role)
+  end
+
+  def in_finance_group?
+    %w[finance finance_admin finance_super].include?(role)
+  end
+
+  # Permissions
+
+  def ability
+    @ability ||= Ability.new(self)
+  end
+
+  delegate :can?, :cannot?, to: :ability
```

Comment:
[NIT] Can we move this bit to the top of the class with the rest of the class methods?